crates/oxc_query/src/vertex.rs

```diff
@@ -278,6 +278,29 @@ impl<'a> Vertex<'a> {
         }
     }
 
+    pub fn function_scope_flag<'c, 'b>(&self, adapter: &'c Adapter<'b>) -> ScopeFlags {
+        let wanted_node_hash = match &self {
+            Vertex::ArrowFunction(data) => calculate_hash(data.arrow_expression),
+            Vertex::FnDeclaration(data) => calculate_hash(data.function),
+            _ => unreachable!(
+                "'function_scope_flag' function should only ever be called with an ArrowFunction or FnDeclaration"
+            ),
+        };
+
+        let found = adapter.semantic.nodes().iter().find(|x| {
```

Comment:
it's not great to iterate the entire node stack each time, is it worth adding a hashmap here to just store the calculated hash to the scope flags so if we do multiple lookups we don't need to iterate all the nodes multiple times?

Reply:
Let's just iterate it until performance regression is noticeable.